ui/apps/platform/src/utils/searchUtils.ts

```diff
@@ -350,6 +350,23 @@ export function hasSearchKeyValue(search: string, key: string, value: string | n
     return urlSearchParams.get(key) === value || urlSearchParams.get(key) === encodedValue;
 }
 
+/**
+ * Finds a value in an object by key case-insensitively.
+ *
+ * @param obj The object to search in
+ * @param targetKey The key to search for (case-insensitive)
+ * @returns The value associated with the key, or undefined if not found
+ */
+export function getValueByCaseInsensitiveKey<T extends Record<string, unknown>>(
```

Comment:
Fun! This might be useful in other VM pages too since we have some issues with case-sensitive keys there as well.